src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -19,16 +19,30 @@ public static Stream GetStream(this IPipeConnection connection)
 
     public static class PipelineWriterExtensions
     {
+        private readonly static Task _completedTask = Task.FromResult(0);
+
         public static Task WriteAsync(this IPipeWriter output, Span<byte> source)
         {
             var writeBuffer = output.Alloc();
             writeBuffer.Write(source);
-            return AwaitFlushAsync(writeBuffer);
+            return FlushAsync(writeBuffer);
+        }
+
+        private static Task FlushAsync(WritableBuffer writeBuffer)
+        {
+            var awaitable = writeBuffer.FlushAsync();
+            if (awaitable.IsCompleted)
+            {
+                awaitable.GetResult();
```

Comment:
> Throw an exception (if there was any)

Gah wonder if this is related https://github.com/aspnet/KestrelHttpServer/issues/1435; might need to return a faulted Task... :-/